posthog/cloud_utils.py

```diff
@@ -38,3 +44,45 @@ def is_cloud():
 def TEST_clear_cloud_cache(value: Optional[bool] = None):
     global is_cloud_cached
     is_cloud_cached = value
+
+
+def get_cached_instance_license() -> Optional["License"]:
+    """Returns the first valid license and caches the value for the lifetime of the instance, as it is not expected to change.
+    If there is no valid license, it returns None.
+    """
+    global instance_license_cached
+    global is_instance_licensed_cached
+
+    try:
+        from ee.models.license import License
+
+        if isinstance(instance_license_cached, License):
+            return instance_license_cached
+
+        if is_instance_licensed_cached is False:
+            # This is an unlicensed instance
+            return None
+
+        # TRICKY - The license table may not exist if a migration is running
+        license = License.objects.first_valid()
+        if license:
+            instance_license_cached = license
+            is_instance_licensed_cached = True
+        return instance_license_cached
+
+    except ProgrammingError:
```

Comment:
This can only be raised by `License.objects.first_valid()`, right? Best to only wrap the potentially offending lines in a `try:except:`, makes it easier to reason about the code

Reply:
The `try` is actually for the License import on line 59, which is why this starts just above that. The import is required for the `isinstance` check. I'll just wrap the import.